hoomd/pytest/test_box.py

```diff
@@ -0,0 +1,201 @@
+from copy import copy
+from math import isclose
+import numpy as np
+from pytest import fixture, raises
+
+from hoomd.box import Box
+
+
+@fixture
+def box_dict():
+    return dict(Lx=1, Ly=2, Lz=3, xy=1, xz=2, yz=3)
+
+
+def test_base_constructor(box_dict):
+    box = Box(**box_dict)
+    for key in box_dict:
+        assert getattr(box, key) == box_dict[key]
+
+
+@fixture
+def base_box(box_dict):
+    return Box(**box_dict)
+
+
+def test_cpp_python_correspondence(base_box):
+    cpp_obj = base_box._cpp_obj
+    cpp_L = cpp_obj.getL()
+    assert base_box.Lx == cpp_L.x and base_box.Ly == cpp_L.y \
+        and base_box.Lz == cpp_L.z
+    assert base_box.xy == cpp_obj.getTiltFactorXY()
+    assert base_box.xz == cpp_obj.getTiltFactorXZ()
+    assert base_box.yz == cpp_obj.getTiltFactorYZ()
+
+
+def test_setting_Ls(base_box):
+    for attr in ['Lx', 'Ly', 'Lz']:
+        for L in np.linspace(1, 100, 10):
+            setattr(base_box, attr, L)
+            assert getattr(base_box, attr) == L
+    for L in np.linspace(1, 100, 10):
+        base_box.L = L
+        assert all(base_box.L == L)
+    base_box.L = [3, 2, 1]
+    assert all(base_box.L == [3, 2, 1])
+
+
+def test_setting_tilts(base_box):
+    for attr in ['xy', 'xz', 'yz']:
+        for tilt in np.linspace(1, 100, 10):
+            setattr(base_box, attr, tilt)
+            assert getattr(base_box, attr) == tilt
+    for tilt in np.linspace(1, 100, 10):
+        base_box.tilts = tilt
+        assert all(base_box.tilts == tilt)
+    base_box.tilts = [3, 2, 1]
+    assert all(base_box.tilts == [3, 2, 1])
+
+
+def test_is2D(base_box):
+    base_box.Lz = 0
+    assert base_box.is2D
+    for L in np.linspace(1, 100, 10):
+        base_box.Lz = L
+        assert not base_box.is2D
+
+
+def test_dimensions(base_box):
+    base_box.Lz = 0
+    assert base_box.dimensions == 2
+    for L in np.linspace(1, 100, 10):
+        base_box.Lz = L
+        assert base_box.dimensions == 3
+
+
+def test_lattice_vectors(base_box):
```

Comment:
There should be a warning and test for passing in a left-handed box matrix.

Reply:
We don't allow setting lattice vectors directly. The matrix property does check that the expected values are zero.